receipt_upload/receipt_upload/merchant_resolution/resolver.py

```
@@ -431,10 +431,43 @@ def _resolve_impl(
             merchant_hint = merchant_line.text if merchant_line else None
 
+        # The full text of the line(s) carrying the MERCHANT_NAME label. We find
+        # it via the label, NOT geometry: the top-line heuristic is unreliable
+        # (receipts often put the check #, e.g. "105", on the first line, and the
+        # OCR y-axis orientation varies), so it would miss the real name line.
+        merchant_name_line_ids = {
+            lab.line_id for lab in word_labels if lab.label == "MERCHANT_NAME"
+        }
+        merchant_name_line_text = " ".join(
+            ln.text
+            for ln in sorted(lines, key=lambda l: l.line_id)
```

Ruff (0.15.18)

[error] Ambiguous variable name: `l`

(E741)